### PR TITLE
Push Docker images to AWS ECR

### PR DESCRIPTION
**Pull request by AntoineAugusti**
Same PR than https://github.com/cds-snc/notification-admin/pull/845 in admin

Changed the trigger condition to `push` to make sure that the CI job works as intended. Pods in staging run with Docker images pulled from AWS at the moment.

Gallery link: https://gallery.ecr.aws/v6b8u5o6/notify-document-download-api

### CODE REVIEW

File: ci/Dockerfile [hunk 1]
```diff
@@ -2,7 +2,7 @@ FROM python:3.6-alpine
 
 ENV PYTHONDONTWRITEBYTECODE 1
 
-RUN apk add --no-cache build-base git gcc make libffi-dev && rm -rf /var/cache/apk/*
+RUN apk add --no-cache build-base git gcc make libffi-dev openssl-dev && rm -rf /var/cache/apk/*
```
Comment (brdunfield) [inline ci/Dockerfile:5]: Is this change related to the rest of the PR?

**Comment by brdunfield** [comment]
This looks like it's also addressing the dependency issues?

**Comment by AntoineAugusti** [comment]
Yeah bundled dependencies in this PR because it blocked my PR, I hope that's fine with you.

I had to require `openssl-dev` in the Dockerfile when upgrading `eventlet`. You had to do a similar change on the admin a while ago https://github.com/cds-snc/notification-admin/pull/697#issuecomment-667237295

File: ci/Dockerfile (hunk 1 above)
Comment (AntoineAugusti) [reply ci/Dockerfile:5]: yes, replied in a comment

File: .github/workflows/docker.yaml [hunk 2]
```diff
@@ -1,38 +1,60 @@
...
       env:
         AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
         AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
-      with:
-        args: set image deployment.apps/document-download-api document-download-api=gcr.io/cdssnc/notify/document-download-api:${GITHUB_SHA::7} -n=notification-canada-ca --kubeconfig=/github/home/.kube/config
+      run: |
+        kubectl set image deployment.apps/document-download-api document-download-api=$DOCKER_SLUG:${GITHUB_SHA::7} -n=notification-canada-ca --kubeconfig=$HOME/.kube/config
+    - uses: cds-snc/notification-pr-bot@master
+      env:
+        TOKEN: ${{ secrets.PR_BOT_TOKEN }}
```
Comment (brdunfield) [inline .github/workflows/docker.yaml:60]: Is the PR bot a new addition here?

Comment (AntoineAugusti) [reply .github/workflows/docker.yaml:60]: It is, thanks for being careful! The PR bot is not hooked up yet, so now it'll do nothing but this change is required if we want to have document download API + frontend changes in the automated PR

**Review by brdunfield — APPROVED** [review]
🦑 🔍
